### PR TITLE
Replace content in device-management.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Replace old content with links to the Device Management Client example for Mbed OS.

### CODE REVIEW

**Comment by iriark01** [comment]
Why is this content entirely on GitHub? Shouldn't it be on one of the docs sites?

**Comment by MarceloSalazar** [comment]
I agree with Irit. It would be good to have the content rendered in the docs site.
I'd like to keep the source of the content in the existing location in github (or possibly split to another .md file if that helps).

**Comment by MarceloSalazar** [comment]
Please review IOTWEB-5909
We may need to remove the content altogether.

**Comment by AnotherButler** [comment]
@iriark01 This is now part of a product discussion.

**Review by MarceloSalazar — APPROVED** [review]
This looks good and hope we can merge it asap.

The current content related to device management in the mbed.com docs site is very old and should be removed asap (as part of this PR).

**Comment by iriark01** [comment]
I'm still trying to understand why we're sending people to GitHub instead of pulling this content into doc sites - either Mbed OS or DM.

There is a lot to discuss around DM - we also have Update content and Mbed CLI content that needs handling.

@JanneKiiskila is this content going on the DM site, since it's Client based?

**Review by JanneKiiskila — APPROVED** [review]
Please, let's get this in.
GitHub is a really good place for developers to go... :-)

**Comment by iriark01** [comment]
Documentation is what we have docs sites for. Why are we now keeping content only on GitHub?

**Comment by JanneKiiskila** [comment]
We can then of course link to this; https://www.pelion.com/docs/device-management/current/welcome/index.html -> and then the users will be having an interesting journey trying to find the examples from there.

**Comment by JanneKiiskila** [comment]
Going forwards, I think there will be a section for AWS IoT, Google IoT, Azure IoT etc...

**Comment by iriark01** [comment]
My question was: why are you not adding this content to the docs site? Why keep it only on GitHub?

**Comment by cvasilak** [comment]
I guess as a temporary workaround (until IOTWEB-5909 handles it) and not to confuse users with **deprecated content that is currently online**, my suggestion is to update the links in this PR to point to Pelion documentation instead of GitHub as @iriark01 suggested.

That is:

The example's [developer guide] should point to: https://www.pelion.com/docs/device-management/current/connecting/tutorial-pelion-mbedos.html

The example's [porting guide] should point to:  https://www.pelion.com/docs/device-management/current/porting/index.html

I can open PR against this if we agree.

@AnotherButler @iriark01 WDYT?

**Comment by iriark01** [comment]
Pointing to the docs site is good, but I'm still waiting for Product approval for this. I've spoken to Andy and hope to have a final answer today.

File: docs/ref-designs/pelion_ready/device-management.md
Comment (iriark01) [inline docs/ref-designs/pelion_ready/device-management.md]:
```suggestion
- The example's [developer guide](https://www.pelion.com/docs/device-management/current/connecting/tutorial-pelion-mbedos.html).
```

File: docs/ref-designs/pelion_ready/device-management.md
Comment (iriark01) [inline docs/ref-designs/pelion_ready/device-management.md]:
```suggestion
- The example's [porting guide](https://www.pelion.com/docs/device-management/current/porting/index.html).
```

**Review by iriark01 — COMMENTED** [review]
Replacing links. 
I'm going to use this PR for a few other changes related to DM content on Mbed OS, as agreed on Jira.

**Comment by iriark01** [comment]
Merging into 6; will need to manually handle in 5.15 as it's too different for a simple cherry pick

**Pull request merged**
